quickwit/quickwit-control-plane/src/indexing_scheduler/scheduling/mod.rs

```diff
@@ -124,7 +125,19 @@ fn convert_physical_plan_to_solution(
                     source_id: indexing_task.source_id.clone(),
                 };
                 if let Some(source_ord) = id_to_ord_map.source_ord(&source_uid) {
-                    indexer_assignment.add_shards(source_ord, indexing_task.shard_ids.len() as u32);
+                    let source_type = &sources[source_ord as usize].source_type;
+                    match source_type {
+                        SourceToScheduleType::Sharded { .. } => {
+                            indexer_assignment.add_shards(source_ord, indexing_task.shard_ids.len() as u32);
+                        }
+                        SourceToScheduleType::NonSharded { .. } => {
+                            // For non-sharded sources like Kafka, one pipeline = one shard in the solutions
+                            indexer_assignment.add_shards(source_ord, 1);
```

Comment:
the conversion for non sharded sources (kafka) was incorrect.

Reply:
In the NonSharded source, you have a `num_pipelines`, should we not use it?

Reply:
forget my comment, we are iterating on indexer task here.